crates/wasi/src/preview2/ctx.rs

```diff
@@ -113,11 +113,20 @@ impl WasiCtxBuilder {
         self
     }
 
+    /// Set the generator for the secure random number generator.
+    ///
+    /// This initializes the random number generator using
+    /// [`rand::thread_rng`].
+    pub fn set_random(mut self) -> Self {
+        self.random = Some(random::thread_rng());
+        self
+    }
+
     /// Set the generator for the secure random number generator. This
     /// is only avalabile under `cfg(test)` as it's only intended for use
     /// by tests.
     #[cfg(test)]
-    pub fn set_random(mut self, random: impl RngCore + Send + Sync + 'static) -> Self {
+    pub fn set_random_for_testing(mut self, random: impl RngCore + Send + Sync + 'static) -> Self {
```

Comment:
I'm not sure of the underlying motivations for this, but I would naively expect this to be `set_random` and not `#[cfg(test)]` where the new method added in this PR would be something like `set_random_randomized` (or `set_random_system` or something like that). While the CLI likely doesn't want `set_random_for_testing` I could see that being useful in a deterministic record-and-replay sort of scenario